torch/linalg/__init__.py

```diff
@@ -184,3 +184,94 @@
     >>> torch.allclose(torch.tensordot(a, x, dims=x.ndim), b, atol=1e-6)
     True
 """)
+
+
+qr = _add_docstr(_linalg.linalg_qr, r"""
+qr(input, mode='reduced', *, out=None) -> (Tensor, Tensor)
+
+Computes the QR decomposition of a matrix or a batch of matrices :attr:`input`,
+and returns a namedtuple (Q, R) of tensors such that :math:`\text{input} = Q R`
+with :math:`Q` being an orthogonal matrix or batch of orthogonal matrices and
+:math:`R` being an upper triangular matrix or batch of upper triangular matrices.
+
+Depending on the value of :attr:`mode` this function returns the reduced or
+complete QR factorization. See below for a list of valid modes.
+
+.. warning:: **Differences with** :meth:`~torch.qr`:
```

Comment:
I think we can remove this section. We should deprecate torch.qr in the future and not worry about describing the differences.

Reply:
as for `svd`, I think that from the user might wonder why there are two functions with the same name, especially if he/she uses the "search" box in the docs. So being able to immediately compare them is useful.

But I am fine to delete it if you really want.

Reply:
You make a good point and it'd be nice if our docs could put "deprecated" next to the name of the functions when they're searched for, so the search would show something like:

```
torch.linalg.qr
torch.qr (deprecated)
```

Adding the deprecation note to torch.qr will be half the battle. cc @heitorschueroff who's also going to help us review torch.linalg consistency before the 1.8 release. 

Let's delete this for now and consider what we want to do systematically before the 1.8 release. That is, if we do want warnings or notes like this we should put them on all the new linear algebra functions that replace existing functions, too.

Reply:
to be consistent with what we decided to do in #45562 , I move the "differences" box to `torch.qr`, see commit 723c4d5ca2.
However, I noticed that there is a problem with how sphinxs renders ``:meth:`~torch.linalg.qr` ``: see the following pic:
![image](https://user-images.githubusercontent.com/344507/101518536-53d60a00-3982-11eb-94f8-5d5508cd6aa1.png)

as you can see, the reference to `torch.linalg.qr` is rendered as just `qr`. So, as a workaround I prefixed it with a fixed-font ``torch.linalg.``, but it's ugly.

I think that this misbehavior should be fixed in in another PR, though.

Reply:
`:meth:` seems wrong. I would expect it to work right if you do `` `torch.linalg.qr` `` (just single backticks). And otherwise it should be `:func:` right?

Cc @mattip as our resident expert.

Reply:
The single backticks don't work (they are just rendered in *italics*). I also tried to use `:func:` instead of `:meth:` but I get the same rendering as the picture above

Reply:
These kind of deliberations can be resolved by looking at the objects stored in https://github.com/pytorch/pytorch.github.io/blob/site/docs/master/objects.inv (or https://pytorch.org/docs/master/objects.inv) with [sphobjinv](https://github.com/bskinn/sphobjinv) like this:
```
$ pip install sphobjinv
$ wget https://pytorch.org/docs/master/objects.inv
$ sphobjinv suggest objects.inv linalg.qr
:py:method:`torch.Tensor.qr`
:py:module:`torch.linalg`
:py:function:`torch.linalg.cholesky`
:py:function:`torch.linalg.cond`
:py:function:`torch.linalg.det`
:py:function:`torch.linalg.eigh`
:py:function:`torch.linalg.eigvalsh`
:py:function:`torch.linalg.matrix_rank`
:py:function:`torch.linalg.norm`
:py:function:`torch.linalg.tensorinv`
:py:function:`torch.linalg.tensorsolve`
:py:function:`torch.qr`
:std:doc:`generated/torch.qr`
:std:label:`generated/torch.qr:torch.qr`
:std:doc:`linalg`
:std:label:`linalg:functions`
:std:label:`linalg:torch.linalg`
```